src/utils/entitySpec.js

```diff
@@ -180,4 +180,23 @@ export default class EntitySpec {
   toString() {
     return this.toPath()
   }
+
+  static getRevisionToKey(revision, definition) {
+    return definition.provider === 'github' ? revision.sha : revision
```

Comment:
`revision` should always be a string.  Feels like this method and `getRevisionToString` are helpers for some specific scenarios and are just being dumped on this class. Suggest they go somewhere else as they do not match the nature of EntitySpec itself.

Reply:
I've moved these methods into the Definition class.